src/main/flight/gyroanalyse.c

```diff
@@ -279,6 +322,7 @@ static FAST_CODE_NOINLINE void gyroDataAnalyseUpdate(gyroAnalyseState_t *state,
                     if (!fftIncreased) {
                         binStart = i; // first up-step bin
                         fftIncreased = true;
+                        dataMin = state->fftData[i - 1];
```

Comment:
Looks like dataMin is only being set the once here (the very first time a bin is detected that is greater than the previous, ie when !fftIncreased) during the peak detection loop.  Since dataMax:dataMin ratio is important in this new implementation for dynamic control of the PT1 smoothing (deciding what a 'big' peak is), should dataMin be updated as the rest of the bins are checked whenever a new lower min is detected?

As is, I would think that if the low frequency end of the FFT happens to be relatively noisy then it could skew the ratio dataMax:dataMin ratio.  For example, imagine these imaginary bin contents:

A: [7,8,6,1,2,1,1,1,1,1,4,9,3,1,1,1,1...]
vs
B: [1,2,1,7,8,6,1,1,1,1,4,9,3,1,1,1,1...] 

Both of these have the same 3 'spikes':  [1,2,1], [7.8.6], and [4,9,3] with the rest of the bins having the value '1'.  For scenario 'A' the the dataMax:dataMin ratio would be 9:8, and for scenario 'B' it would be 9:1, and ideally I believe you'd want that ratio to be the same in both of the above cases as all the spikes are the same, just re-ordered.

I think probably it should keep checking for lower dataMins throughout the loop, something like this at the same level as the 'if (!fftIncreased)'?:

if (state->fftData[i - 1] < dataMin) {
    dataMin = state->fftData[i - 1];
}